### PR TITLE
Add withSeqEffToIO and withConcEffToIO functions

### PR DESCRIPTION
**Pull request by Nexmean**
address issue https://github.com/haskell-effectful/effectful/issues/112

### CODE REVIEW

**Comment by arybczak** [comment]
Thanks. I'd prefer the function with a signature

```
???
  :: (HasCallStack, IOE :> es)
  => UnliftStrategy
  -> ((forall r. Eff es r -> IO r) -> IO a)
  -> Eff es a
```

though for consistency. But I'm not sure about its name :thinking: `withEffToIO'` in theory works, but I really dislike function variations with ticks since they're non-descriptive.

**Comment by Nexmean** [comment]
I think that common case for this functions is to implement some effect that need to provide callback to library that use IO. So there you don't need to send to your functions UnliftStrategy as argument. And it will be cleaner to call just `withSeqEffToIO` than call `withUnliftStrategyEffToIO SeqUnlift`.

**Comment by arybczak** [comment]
You seem to have a particular use-case in mind, can you show me?

**Comment by Nexmean** [comment]
For example, I use `hw-kafka-client` and I need to provide log callback to it:
```haskell
getKafkaLoggerCallback :: (IOE :> es, Logger :> es) => Eff es Kafka.Callback
getKafkaLoggerCallback = withConcEffToIO Ephemeral Unlimited \effToIO ->
  pure $ Kafka.logCallback \level facility message ->
    effToIO
    $ addLoggingNamespace "kafka"
    $ addLoggingContext (sl "kafka_facility" facility)
    $ logWithLevel (fromKafkaLogLevel level)
    $ fromString message
```
Or when I use `servant-client` I need to provide callback for logging request and response, but with `SeqUnlift` strategy.

Or when I use `Resource` effect I need logging at acquiring and closing too.

File: effectful-core/src/Effectful/Internal/Monad.hs [hunk 1]
```diff
@@ -179,9 +181,28 @@ withEffToIO
   -- ^ Continuation with the unlifting function in scope.
   -> Eff es a
 withEffToIO f = unliftStrategy >>= \case
-  SeqUnlift -> unsafeEff $ \es -> seqUnliftIO es f
+  SeqUnlift      -> unsafeEff $ \es -> seqUnliftIO es f
   ConcUnlift p b -> unsafeEff $ \es -> concUnliftIO es p b f
 
+-- | Create an unlifting function with 'SeqUnlift' strategy.
```
Comment (arybczak) [inline effectful-core/src/Effectful/Internal/Monad.hs:187]:
```suggestion
-- | Create an unlifting function with the 'SeqUnlift' strategy.
```

File: effectful-core/src/Effectful/Internal/Monad.hs [hunk 2]
```diff
@@ -179,9 +181,28 @@ withEffToIO
...
+withSeqEffToIO
+  :: (HasCallStack, IOE :> es)
+  => ((forall r. Eff es r -> IO r) -> IO a)
+  -- ^ Continuation with the unlifting function in scope.
+  -> Eff es a
+withSeqEffToIO f = unsafeEff $ \es -> seqUnliftIO es f
+
+-- | Create an unlifting function with 'ConcUnlift' strategy.
```
Comment (arybczak) [inline effectful-core/src/Effectful/Internal/Monad.hs:195]:
```suggestion
-- | Create an unlifting function with the 'ConcUnlift' strategy.
```

**Review by arybczak — COMMENTED** [review]
Ok, that looks reasonable :+1:

**Pull request merged**